src/content/docs/en/reference/api-reference.mdx

```diff
@@ -426,6 +426,25 @@ And would render HTML like this:
 </ul>
 ```
 
+
+### `Astro.locals`
+
+`Astro.locals` is an API available via [Astro middleware](/en/guides/middleware/). It's an object that can contain any arbitrary information.
+
+```astro title="src/pages/Orders.astro"
+---
+const title = Astro.locals.welcomeTitle();
+const orders = Array.from(Astro.locals.orders.entries());
+---
+<h1>{title}</h1>
+<p>This {data.property} is from middleware.</p>
```

Comment:
```suggestion
```
I'm going to suggest we remove this, since we don't show `data` anywhere. I think the two matching examples are enough!

Reply:
I missed that, good catch!